src/components/memorygame/MemoryGameCards/index.jsx

```diff
@@ -0,0 +1,171 @@
+  );
+  // selectedCards is an array of clicked card states(objects).
+  // selectedCards array hold only "NUM_REQUIRED_MATCHES" amount of card states
+  const [selectedCards, setSelectedCards] = useState([]);
+
+  // handle click flips card and adds that card's state to selectedCards
+  const handleClick = (clickedCardState) => {
+    const copyOfCardStates = [...cardStates];
+    // current index is needed to keep the order of cards displayed
+    // because goal is to make user remember the spots of previous cards
+    const currentIndex = cardStates.findIndex(
+      (state) => state.cardKey === clickedCardState.cardKey
+    );
+    copyOfCardStates[currentIndex].isFlipped = true;
+    setCardStates(() => [...copyOfCardStates]);
+    setSelectedCards((prevState) => [...prevState, clickedCardState]);
+  };
+
+  // matched cards should have different cardKey, same image key, isFlipped:true
+  // function below tests it
+  const doesMatchLastCard = (prevCardState, lastCardState) => {
+    return (
+      prevCardState.cardKey !== lastCardState.cardKey &&
+      prevCardState.img.imgKey === lastCardState.img.imgKey &&
+      prevCardState.isFlipped &&
+      lastCardState.isFlipped
+    );
+  };
+
+  // Triggered when user clicks on a card(selectedCards change)
+  // checks selected cards to see if they are a match
+  useEffect(() => {
+    // only triggered if "NUM_REQUIRED_MATCHES" amount of cards are clicked
+    if (selectedCards.length !== NUM_REQUIRED_MATCHES) {
+      return;
+    }
+    // match condition of selectedCards will be based on card that is added last to selectedCards
+    const lastCardState = selectedCards[selectedCards.length - 1];
+    const matchesOfLastCardState = selectedCards.filter((prevCard) =>
+      doesMatchLastCard(prevCard, lastCardState)
+    );
+    // matchesOfLastCardState doesnt include last card
+    const matchedCardsArr = [...matchesOfLastCardState, lastCardState];
+    const copyOfCardStates = [...cardStates];
+    // matchedCards is an array states that holds successfully matched card states
+    if (matchedCardsArr.length === NUM_REQUIRED_MATCHES) {
+      matchedCardsArr.forEach((matchedCard) => {
+        // current index is needed to keep the order of cards displayed
+        const targetIndex = copyOfCardStates.findIndex(
+          (card) => card.cardKey === matchedCard.cardKey
+        );
+        // if isMatched is true card image gets replaced by success symbol
+        copyOfCardStates[targetIndex] = {
+          ...copyOfCardStates[targetIndex],
+          isMatched: true,
+        };
+      });
+      // to display 1 fact per match,
+      // only one of the matched cards(selectedCards[0]) should be passed as state
+      setMatchedCards((prevState) => [...prevState, selectedCards[0]]);
+      setCardStates(() => [...copyOfCardStates]);
+    }
+    // if there is no match or not enough matches, flipped cards get
+    // closed by setting isFlipped to false
```

Comment:
flipped cards reset isFlipped to false

change the second half of the comment

Reply:
PTAL